### ISSUE TITLE
Option `onExit` in `terminal.integrated.persistentSessionReviveProcess` also restores terminal buffer when window is closed and then reopened

### ISSUE THREAD

**Issue by karrtikr** [microsoft/vscode#134062]
Testing #133516

There're two options: `onExitAndWindowClose` and `onExit`, both seem to be working the some way. I'm assuming if I close the Window using `Close Window` command, `onExit` should not apply and only `onExitAndWindowClose` should.

**Comment by meganrogge** [comment]
is it the last window that you are closing? if so, according to the description, it should restore.

**Comment by karrtikr** [comment]
I'm using `Close window` command, which apparently closes every tab and exits VSCode. What is meant by a "window" here?

**Comment by meganrogge** [comment]
meaning if you have several vscode windows open then closing and reopening that window with `onExit` set, it shouldn't restore.. 

but if you have only one vscode window open, then it should

**Comment by karrtikr** [comment]
Ah I see, thanks. Can you tell how do I reopen a window which I just closed?

**Comment by meganrogge** [comment]
I think VS code will automatically open the last window you had open when you next open the app

**Comment by karrtikr** [comment]
> when you next open the app

That's true, but I only have to close one window (which is not the last window) and so the app will still be open. I guess what I'm asking is how do I restore a single window while other windows and the app are still open?

**Comment by meganrogge** [comment]
`onExit` should only restore when 1) all of the windows are closed 2) the last window is closed (not on mac) and reopened
`onExitAndWindowClose` should restore when any window is closed and reopened

**Comment by karrtikr** [comment]
> onExitAndWindowClose should restore when any window is closed and reopened

I'm not sure how to test this behavior as I'm not clear on how to accomplish the "reopen" part here for any window.

**Comment by meganrogge** [comment]
@karrtikr let's say you have 3 folders open in separate windows: A, B, C and have onExitAndWindowClose set. 

close the window with folder C. in window A or B, go to file -> open recent -> open folder C. the terminal buffers should be restored.

@Tyriar correct this if it's inaccurate please
